### ISSUE TITLE
generation with nest-cli Error: Unknown argument skipImport. Did you mean skip-import?

### ISSUE THREAD

**Issue by kaflan** [nestjs/nest-cli#1693]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

Hello, I try to do an example application learned on udemy. 

I try to run cmd
```js
nest g mo src/tasks
```
 I have angular CLI. But I cannot understand why that error.

### Minimum reproduction code

nest g mo src/tasks

### Steps to reproduce

1. nest g mo src/tasks

### Expected behavior

(node:33348) UnhandledPromiseRejectionWarning: Error: Unknown argument skipImport. Did you mean skip-import?
    at parseArgs (/Users/andreykatrusha/Repos/nestjs-task-managment/node_modules/@angular-devkit/schematics-cli/bin/schematics.js:338:19)
    at main (/Users/andreykatrusha/Repos/nestjs-task-managment/node_modules/@angular-devkit/schematics-cli/bin/schematics.js:122:49)
    at Object.<anonymous> (/Users/andreykatrusha/Repos/nestjs-task-managment/node_modules/@angular-devkit/schematics-cli/bin/schematics.js:367:5)
    at Module._compile (internal/modules/cjs/loader.js:1072:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
    at Module.load (internal/modules/cjs/loader.js:937:32)
    at Function.Module._load (internal/modules/cjs/loader.js:778:12)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:76:12)
    at internal/main/run_main_module.js:17:47
(Use `node --trace-warnings ...` to show where the warning was created)
(node:33348) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:33348) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

### Package version

8.2.7

### NestJS version

8.4.7

### Node.js version

14.17.5

### In which operating systems have you tested?

- [X] macOS
- [x] Windows
- [x] Linux

**Comment by kaflan** [comment]
I think maybe it conflict with CLI?

**Comment by kaflan** [comment]
https://github.com/nestjs/nest-cli/issues/323
https://github.com/nestjs/nest-cli/issues/145

**Comment by kaflan** [comment]
> Got the same error with `global NestJS 8.2.7`, project `@nestjs/cli: 8.0.0` while executing a simple `nest g controller Something` on Windows 10.
> 
> The command executed by nest: `node @nestjs/schematics:controller --name=something --no-dry-run --no-skipImport --language="ts" --sourceRoot="src" --spec`

on my computer I have
```
node @nestjs/schematics:controller --name=something --no-dry-run --no-skipImport --language="ts" --sourceRoot="src" --spec
internal/modules/cjs/loader.js:892
  throw err;
  ^

Error: Cannot find module '/Users/andreykatrusha/Repos/nestjs-task-managment/@nestjs/schematics:controller'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:889:15)
    at Function.Module._load (internal/modules/cjs/loader.js:745:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:76:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

**Comment by kaflan** [comment]
<img width="989" alt="Screenshot 2022-06-23 at 17 26 46" src="https://user-images.githubusercontent.com/3959504/175323342-1401c466-75d4-4a9a-ab4b-6c20a9ccea56.png">

**Comment by micalevisk** [comment]
@kaflan I guess that's another issue.

---

The original one is a bug on v8.2.7.

you can downgrade it for now. v8.2.6 of `@nestjs/cli` worked just fine. I don't recommend using Nest CLI globally (use `yarn nest` or `npx nest`).

**Comment by kaflan** [comment]
When I try to install https://www.npmjs.com/package/@nestjs/schematics
I have that error

**Comment by kaflan** [comment]
> 8.2.6

Ok I  try now

**Comment by BhargavMantha** [comment]
I can confirm this issue as well
environment:

System:
  Kernel: 5.15.46-1-MANJARO arch: x86_64 bits: 64 compiler: gcc v: 12.1.0
  Desktop: KDE Plasma v: 5.24.5 tk: Qt v: 5.15.4 wm: kwin_x11 vt: 1 dm: SDDM
  Distro: Manjaro Linux base: Arch Linux

### Quick solution:

1. add this to your nest-cli.json
```
"root": "src"
```

2. change into the src of your project folder
```
cd src
```

3. run 
```
nest generate module messages
```

if the above does not work also try changing the version of packages in package.json to 
```
"@nestjs/cli": "^8.2.6",
"@nestjs/schematics": "^8.0.11",
```
removing the node modules in the current project 
reinstalling the nest cli with
```
npm uninstall -g @nestjs/cli
npm i -g @nestjs/cli@8.2.6 
```

**Comment by kaflan** [comment]
Add sistem

**Comment by jmcdo29** [comment]
@micalevisk @kamilmysliwiec Looks like this was a breaking change of [`@angular-devkit/schematics@14.0.0`](https://github.com/angular/angular-cli/blob/HEAD/CHANGELOG.md#angular-devkitschematics-cli). We should revert the angular schematics packages until we're ready to go through the full upgrade to keep from making breaking changes in our shcematics

**Comment by Acetyld** [comment]
Same issue here., tried above soluions without succes

**Comment by micalevisk** [comment]
@Acetyld `npm i @nestjs/cli@8.2.6` and then `npx nest g mo foo` will work for sure. If didn't worked for you, then you're probably missing something. We could help you on Discord.


![image](https://user-images.githubusercontent.com/13461315/175359768-88b0f4ef-d77c-4b08-a9e1-49fff42375b4.png)

**Comment by kaflan** [comment]
Try tommorow

**Comment by seaweasel** [comment]
Bump. Having the same problem.

**Comment by jmcdo29** [comment]
Guys, we know the problem, we know you're having it, no need to "bump". This literally came up today, it's not like this is inactive in any way.

**Comment by kamilmysliwiec** [comment]
Fixed in 8.2.8
